common/src/main/java/io/novafoundation/nova/common/view/TipsInputField.kt

```diff
@@ -0,0 +1,171 @@
+package io.novafoundation.nova.common.view
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.PointF
+import android.text.method.DigitsKeyListener
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.View
+import android.widget.EditText
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.view.isVisible
+import io.novafoundation.nova.common.R
+import io.novafoundation.nova.common.utils.WithContextExtensions
+import io.novafoundation.nova.common.utils.onTextChanged
+import io.novafoundation.nova.common.utils.setImageTintRes
+import io.novafoundation.nova.common.utils.useAttributes
+import io.novafoundation.nova.common.view.shape.getInputBackground
+import io.novafoundation.nova.common.view.shape.getInputBackgroundError
+import kotlinx.android.synthetic.main.view_tips_input.view.tipsInputClear
+import kotlinx.android.synthetic.main.view_tips_input.view.tipsInputContainer
+import kotlinx.android.synthetic.main.view_tips_input.view.tipsInputError
+import kotlinx.android.synthetic.main.view_tips_input.view.tipsInputField
+import kotlinx.android.synthetic.main.view_tips_input.view.tipsInputFieldContainer
+
+class TipsInputField @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : LinearLayout(context, attrs, defStyle), WithContextExtensions by WithContextExtensions(context) {
+
+    private var postfix: String? = null
+    private var postfixPadding = 4.dp
+    private var postfixPosition: PointF? = null
+
+    private val textPaint: Paint
+
+    val content: EditText
+        get() = tipsInputField
+
+    init {
+        orientation = VERTICAL
+        View.inflate(context, R.layout.view_tips_input, this)
+        tipsInputFieldContainer.setAddStatesFromChildren(true)
+        tipsInputFieldContainer.background = context.getInputBackground()
+
+        content.onTextChanged {
+            tipsInputClear.isVisible = it.isNotEmpty()
+            tipsInputContainer.isVisible = it.isEmpty()
+            measurePostfix(it)
+            invalidate()
+        }
+
+        textPaint = Paint(content.paint).apply {
+            color = content.currentTextColor
```

Comment:
Doesn't `Paint(content.paint)` also copy the color?

Reply:
Yes it does. I'll fix it